cl/corpus_importer/management/commands/import_tn.py

```diff
@@ -0,0 +1,188 @@
+    """
+
+    judges = case["judge"]
+    first_name = judges.split(" ")[0].title()
+
+    lead_author = Person.objects.get(
+        positions__court_id=case["court"],
+        is_alias_of=None,
+        name_first=first_name,
+    )
+    # Four Panelists exist, one retired end of 2019 and the other joined.
+    panelists = case["judge"]
+    if case["court"] == "tennworkcompapp":
+        exclude = (
+            "Marshall"
+            if parser.parse(case["pub_date"]).date().year == 2020
+            else "Pele"
+        )
+        panelists_query = Person.objects.filter(
+            positions__court_id=case["court"]
+        ).exclude(name_first=exclude)
+        panelists = ", ".join([x.name_full for x in panelists_query])
+
+    return {
+        "source": Docket.DIRECT_INPUT,
+        "cluster_source": "D",
+        "case_names": case["title"],
+        "case_dates": parser.parse(case["pub_date"]),
+        "precedential_statuses": "Published",
+        "docket_numbers": case["docket"],
+        "judges": panelists,
+        "author_id": lead_author.id,
+        "author": lead_author,
+        "date_filed_is_approximate": False,
+        "blocked_statuses": False,
+        "neutral_citations": case["neutral_citation"],
+        "download_urls": case["pdf_url"],
+    }
+
+
+def add_neutral_citations(filepath):
+    """Add neutral citations to our dataset
+
+    :param tn_corpus: The case data.
+    :type: dict
+    :return: Our case data with neutral citations included.
+    :type: dict
+    """
+    tn_corpus = json.loads(open(filepath, "r").read())
+    results = []
+    for court in ["tennworkcompapp", "tennworkcompcl"]:
+        reporter = "TN WC App." if court == "tennworkcompapp" else "TN WC"
+        tn = [
+            x
+            for x in sorted(
+                tn_corpus, key=lambda x: (x["pub_date"], x["label"])
+            )
+            if x["court"] == court
+        ]
+        for key, group in itertools.groupby(tn, lambda x: x["pub_date"][:4]):
+            count = 1
+            for case in list(group):
+                case["neutral_citation"] = "%s %s %s" % (key, reporter, count)
+                results.append(case)
+                count += 1
+    return results
+
+
+def import_tn_corpus(log, skip_until, dir):
+    """Import TN Corpus
+
+    :param log: Should we view logging info
+    :param skip_until: Label ID, if any, to process first
+    :param dir: Location of our import files
+    :return: None
+    """
+    ready = False if skip_until else True
+
+    if log:
+        logging.getLogger().setLevel(logging.INFO)
+
+    logging.info("Starting import")
+    filepath = "%s/data.json" % dir
+    tn_corpus = add_neutral_citations(filepath)
+    if not ready:
+        case = [x for x in tn_corpus if x["label"] == int(skip_until)][0]
+        logging.info(
+            "Skipping until case %s labeled: %s", case["title"], case["label"]
+        )
+
+    for case in sorted(tn_corpus, key=lambda x: x["label"]):
```

Comment:
Seems like we should have the data sorted by the function I suggested above, before we get here.

Reply:
√